src/cli/commands/force/lightning/local/device/create.ts

```diff
@@ -0,0 +1,258 @@
+                    'reqs:devicetype:unfulfilledMessage'
+                )
+            }
+        ];
+        this.addRequirements(extraReqs);
+
+        return this.validateInputsParameters() // first validate input parameters
+            .then(() => super.run()) // then validate setup requirements
+            .then(() => {
+                // then execute the create command
+                logger.info(
+                    'Setup requirements met, continuing with Device Create'
+                );
+                return this.executeDeviceCreate();
+            })
+            .then(() => {
+                const tree = cli.tree();
+                const message = util.format(
+                    messages.getMessage('success'),
+                    this.deviceName,
+                    this.deviceType
+                );
+                tree.insert(message);
+                tree.display();
+            })
+            .catch((error) => {
+                logger.warn(`Device Create failed for ${platform}.`);
+                return Promise.reject(error);
+            });
+    }
+
+    public async validateInputsParameters(): Promise<void> {
+        const deviceName = this.flags.devicename as string;
+        const deviceType = this.flags.devicetype as string;
+
+        // ensure that the platform flag value is valid
+        if (!CommandLineUtils.platformFlagIsValid(this.flags.platform)) {
+            return Promise.reject(
+                new SfdxError(
+                    messages.getMessage(
+                        'error:invalidPlatformFlagsDescription'
+                    ),
+                    'lwc-dev-mobile',
+                    this.examples
+                )
+            );
+        }
+
+        // ensure that the device name flag value is valid
+        if (deviceName == null || deviceName.trim() === '') {
+            return Promise.reject(
+                new SfdxError(
+                    messages.getMessage(
+                        'error:invalidDeviceNameFlagsDescription'
+                    ),
+                    'lwc-dev-mobile',
+                    this.examples
+                )
+            );
+        }
+
+        // ensure that the device type flag value is valid
+        if (deviceType == null || deviceType.trim() === '') {
+            return Promise.reject(
+                new SfdxError(
+                    messages.getMessage(
+                        'error:invalidDeviceTypeFlagsDescription'
+                    ),
+                    'lwc-dev-mobile',
+                    this.examples
+                )
+            );
+        }
+
+        this.platform = this.flags.platform;
+        this.deviceName = deviceName;
+        this.deviceType = deviceType;
+        return Promise.resolve();
+    }
+
+    // check whether device name is unique (i.e a virtual device with the same name doesn't exist already)
+    private async checkUniqueDevice(): Promise<string> {
```

Comment:
Can we change the nomenclature of this to something more around "exists", e.g. `checkDeviceExists`? "Unique" can have some other connotations that could cause confusion.

Reply:
sure, but the purpose of this function is to ensure that no other device with the same name exists (i.e success = does not exist). so I'll rename to `checkDeviceDoesNotExist`

Reply:
Ah yes thanks. I wasn't careful about my scan of the name—just more generally proposing "exists" vs. "unique" nomenclature.

Now doing a less sloppy read of the function and its purpose/flow, I'm wondering if `checkDeviceNameAvailable()` might be a more accurate descriptor. I'm pretty sure I'm way overthinking this thing. 😉